src/components/ArticlesPage.jsx

```diff
@@ -1,57 +1,83 @@
 import React from "react";
-import { useState, useEffect } from "react";
-import { getArticles } from "../api";
-import Button from 'react-bootstrap/Button';
-import Card from 'react-bootstrap/Card';
-import { Link } from "react-router-dom";
-import { Spinner } from "react-bootstrap";
+import { useContext, useState } from "react";
+import { LoadingContext } from "../context/LoadingProvider";
+import { Loading } from "./Loading";
+import { ArticleCard } from "./ArticlesCard";
 
-export const ArticlesPage=() => {
-    const [articles, setArticles] = useState([]);
-    const [loading, setLoading] = useState(true);
-  
-    
-  useEffect(() => {
-    getArticles()
-      .then(({articles}) => {
-        setArticles(articles);
-        setLoading(false)
-      })
-      .catch((err) => {console.log(err)});
-  }, []);
+export const ArticlesPage=({articles, setArticles}) => {
+    const {loading, setLoading } = useContext(LoadingContext)
+    const [sortBy, setSortBy] = useState('created');
+    const [order, setOrder] = useState('desc'); 
+
+ // Function to sort articles
+ const sortArticles = (articles, sortBy, order) => {
+    return [...articles].sort((a, b) => {
+      let comparison = 0;
+      if (sortBy === "topic") {
+        comparison = a.topic.localeCompare(b.topic);
+      } else if (sortBy === "author") {
+        comparison = a.author.localeCompare(b.author);
+      } else if (sortBy === "title") {
+        comparison = a.title.localeCompare(b.title);
+      } else if (sortBy === "created") {
+        comparison = new Date(a.created_at) - new Date(b.created_at);
+      } else if (sortBy === "comments") {
+        comparison = a.comment_count - b.comment_count;
+      } else if (sortBy === "votes") {
+        comparison = a.votes - b.votes;
+      }
+
+      return order === "asc" ? comparison : -comparison;
+    });
+  };
```

Comment:
I like the idea of this, but this should not be added the this ticket as it is not relevant and out of scope. This should also ideally be handled by the api instead of done on the frontend.